Initial Camera Port

olijeffers0n opened the pull request:
Closes #52 

This adds the ability to control and render cameras from RustPlus.js

WIP - Docs to come. As a predominantly Java & Python Developer, let me know of any code style issues. The rendering is grim, however this is just something that must be done.

- [x] Review Protobuf Definitions for some type changes.
- [x] Reword Comments on the input types. 
- [x] DeObfuscate decoding logic where possible.

liamcottle commented:
Nice work,

I'll do a proper review of the code, make a few changes and get it merged in. Will likely be a couple of days.

Thanks!

olijeffers0n commented:
Sounds good Liam. I have added an example for the usage and added RateLimit info. The Rendering is not that easy to follow but I am considering it as a "It works, don't question it" type of code block. If you really want I am happy to go through and try and put some meaningful variable names in, but I don't think that really adds much.

liamcottle commented:
Thanks! I've made a few changes and have merged this in. I'll get a new release sorted out later on. 🥳

debagos commented:
Hey nice work, but while testing this I got this error:
```
SyntaxError: Unexpected token '.'
camera.js:398
        return (this.cameraSubscribeInfo?.controlFlags & crosshairControlFlag) === crosshairControlFlag;
                                         ^
```

liamcottle commented:
You’re probably using an old version of nodejs. Try updating. Otherwise
I’ll have to make the code more verbose…

On Tue, 21 Mar 2023 at 5:07 PM, debagos ***@***.***> wrote:

> Hey nice work, but while testing this I got this error:
>
> SyntaxError: Unexpected token '.'
> camera.js:398
>         return (this.cameraSubscribeInfo?.controlFlags & crosshairControlFlag) === crosshairControlFlag;
>                                          ^
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/liamcottle/rustplus.js/pull/53#issuecomment-1477254618>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAYGZTNUMGI3UA4G5KT6DETW5ESRZANCNFSM6AAAAAAV7VDSTQ>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 

*Liam Cottle*

Software Engineer

E: ***@***.***

W: https://liamcottle.com

GPG Key: 0xA9215868C57827F8 <https://liamcottle.com/liamcottle-C57827F8.asc>

[image: twitter] <https://twitter.com/liamcottle>

[image: linkedin] <https://www.linkedin.com/in/liamcottle/>

[image: github] <https://github.com/liamcottle>

debagos commented:
Oh that was indeed the case, thank you @liamcottle :+1:
